gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/visitor/OptimizingVisitor.java

```diff
@@ -155,7 +155,20 @@ public QueryNode visit(SortAndPaginateNode sortAndPaginateNode) {
                               .addAllChildFilter(filterList)
                               .build();
                     }
-                    return new DataFetcherNode(stringListEntry.getKey(), filter);
+                    // There should always be at least one entry
+                    if (!stringListEntry.getValue().isEmpty()) {
+                      String source = stringListEntry.getKey();
+                      DataFetcherNode dataFetcherNode = (DataFetcherNode) stringListEntry.getValue().get(0);
```

Comment:
Trying to understand why this special casing is done? What changes led to this?

Reply:
It was earlier creating a new `DataFetcherNode` which is fine, but since we have now added `limit`, `offset` and `orderBy` in DataFetcherNode, we would need access to `limit`, `offset` and `orderBys`, so that a new DataFetcherNode can be created with those values